### PR TITLE
Simplify webassembly targets

### PR DESCRIPTION
**Pull request by akoeplinger**
Add a subset for compiling only the wasm runtime: `mono.wasmruntime`
This allows us to rebuild the wasm runtime without building the full libraries subset.

Updated documentation with new guidance.

Updated how we build the dotnet.timezones.blat.

### CODE REVIEW

File: docs/workflow/testing/libraries/testing-wasm.md [hunk 1]
```diff
@@ -25,11 +25,11 @@ PATH=/Users/<your_user>/.jsvu/:$PATH V8
 
 Now we're ready to build everything for WebAssembly (for more details, please read [this document](../../building/libraries/webassembly-instructions.md#building-everything)):
 ```bash
-./build.sh --arch wasm --os Browser -c release
+./build.sh -os Browser -c release
```
Comment (ViktorHofer) [inline docs/workflow/testing/libraries/testing-wasm.md:28]:
```suggestion
./build.sh -os Browser -c Release
```

File: docs/workflow/testing/libraries/testing-wasm.md [hunk 2]
```diff
@@ -25,11 +25,11 @@ PATH=/Users/<your_user>/.jsvu/:$PATH V8
...
 ```
 and even run tests one by one for each library:
 ```
-./build.sh --subset libs.tests -t --arch wasm --os Browser -c release
+./build.sh --subset libs.tests --test -os Browser -c release
```
Comment (ViktorHofer) [inline docs/workflow/testing/libraries/testing-wasm.md:32]:
```suggestion
./build.sh libs.tests -test -os Browser -c Release
```

File: src/mono/wasm/wasm.proj [hunk 3]
```diff
@@ -1,13 +1,30 @@
...
   <Target Name="CheckEnv">
     <Error Condition="'$(TargetArchitecture)' != 'wasm'" Text="Expected TargetArchitecture==wasm, got '$(TargetArchitecture)'"/>
     <Error Condition="'$(TargetOS)' != 'Browser'" Text="Expected TargetOS==Browser, got '$(TargetOS)'"/>
     <Error Condition="'$(EMSDK_PATH)' == ''" Text="The EMSDK_PATH environment variable should be set pointing to the emscripten SDK root dir."/>
+    <Error Condition="!Exists('$(MicrosoftNetCoreAppRuntimePackRidLibTfmDir)')" Text="The 'libs' subset must be built before building this project. Missing artifacts: $(MicrosoftNetCoreAppRuntimePackRidLibTfmDir)" />
   </Target>
 
   <ItemGroup>
     <PackageReference Include="$(MicrosoftNETCoreRuntimeICUTransportPackage)" PrivateAssets="all" Version="$(MicrosoftNETCoreRuntimeICUTransportVersion)" GeneratePathProperty="true" />
+    <PackageReference Include="System.Runtime.TimeZoneData" PrivateAssets="all" Version="$(SystemRuntimeTimeZoneDataVersion)" GeneratePathProperty="true" />
```
Comment (ViktorHofer) [inline src/mono/wasm/wasm.proj:27]: PackageDownload would make more sense here so that nuget doesn't check for compatibility. The only downside is that you would need to construct the path to the package yourself which I guess should be fine.

File: src/mono/wasm/wasm.proj [hunk 4]
```diff
@@ -1,13 +1,30 @@
...
+    <MicrosoftNetCoreAppRuntimePackRidLibTfmDir>$([MSBuild]::NormalizeDirectory('$(MicrosoftNetCoreAppRuntimePackRidDir)', 'lib', '$(NetCoreAppCurrent)'))</MicrosoftNetCoreAppRuntimePackRidLibTfmDir>
+    <MicrosoftNetCoreAppRuntimePackNativeDir>$([MSBuild]::NormalizeDirectory('$(MicrosoftNetCoreAppRuntimePackRidDir)', 'native'))</MicrosoftNetCoreAppRuntimePackNativeDir>
+  </PropertyGroup>
 
   <Target Name="CheckEnv">
     <Error Condition="'$(TargetArchitecture)' != 'wasm'" Text="Expected TargetArchitecture==wasm, got '$(TargetArchitecture)'"/>
     <Error Condition="'$(TargetOS)' != 'Browser'" Text="Expected TargetOS==Browser, got '$(TargetOS)'"/>
     <Error Condition="'$(EMSDK_PATH)' == ''" Text="The EMSDK_PATH environment variable should be set pointing to the emscripten SDK root dir."/>
+    <Error Condition="!Exists('$(MicrosoftNetCoreAppRuntimePackRidLibTfmDir)')" Text="The 'libs' subset must be built before building this project. Missing artifacts: $(MicrosoftNetCoreAppRuntimePackRidLibTfmDir)" />
```
Comment (ViktorHofer) [inline src/mono/wasm/wasm.proj:22]: are the lib files only an input or also an output? If only an input then I would probably retrieve them via the livebuild target in livebuilds.targets.

File: src/mono/wasm/wasm.proj (hunk 3 above)
Comment (akoeplinger) [reply src/mono/wasm/wasm.proj:27]: PackageReference is what we use in a lot of other places for this, e.g. https://github.com/dotnet/runtime/blob/master/src/libraries/Common/tests/TestUtilities.Unicode/TestUtilities.Unicode.csproj#L19-L20

The package is guaranteed to not have TFM-specific files so nuget compatibility check is fine, I prefer that to manually constructing path to nuget packages root.

File: src/mono/wasm/Directory.Build.props [hunk 5]
```diff
@@ -0,0 +1,3 @@
+<Project>
```
Comment (ViktorHofer) [inline src/mono/wasm/Directory.Build.props:1]: why are those necessary? MSBuild searches through the dir tree to find the nearest D.B.props & targets.

Comment (akoeplinger) [reply src/mono/wasm/Directory.Build.props:1]: ah yeah this was from an earlier iteration where I didn't use the `Microsoft.Build.NoTargets` in wasm.proj, will remove.

File: src/mono/wasm/wasm.proj (hunk 4 above)
Comment (akoeplinger) [reply src/mono/wasm/wasm.proj:22]: they are only an input, will try replacing with livebuilds.targets

File: src/mono/wasm/wasm.proj (hunk 3 above)
Comment (ViktorHofer) [reply src/mono/wasm/wasm.proj:27]: If it works, sure.

File: src/mono/wasm/wasm.proj [hunk 6]
```diff
@@ -1,13 +1,30 @@
...
+    <CoreLibProject>$([MSBuild]::NormalizePath('$(MonoProjectRoot)', 'netcore', 'System.Private.CoreLib', 'System.Private.CoreLib.csproj'))</CoreLibProject>
+    <PackageRID>browser-wasm</PackageRID>
+    <NativeBinDir>$([MSBuild]::NormalizeDirectory('$(ArtifactsBinDir)', 'native', '$(NetCoreAppCurrent)-$(TargetOS)-$(Configuration)-$(TargetArchitecture)'))</NativeBinDir>
+    <MicrosoftNetCoreAppRuntimePackDir>$([MSBuild]::NormalizeDirectory('$(ArtifactsBinDir)', 'microsoft.netcore.app.runtime.$(PackageRID)', '$(Configuration)'))</MicrosoftNetCoreAppRuntimePackDir>
+    <MicrosoftNetCoreAppRuntimePackRidDir>$([MSBuild]::NormalizeDirectory('$(MicrosoftNetCoreAppRuntimePackDir)', 'runtimes', '$(PackageRID)'))</MicrosoftNetCoreAppRuntimePackRidDir>
+    <MicrosoftNetCoreAppRuntimePackRidLibTfmDir>$([MSBuild]::NormalizeDirectory('$(MicrosoftNetCoreAppRuntimePackRidDir)', 'lib', '$(NetCoreAppCurrent)'))</MicrosoftNetCoreAppRuntimePackRidLibTfmDir>
+    <MicrosoftNetCoreAppRuntimePackNativeDir>$([MSBuild]::NormalizeDirectory('$(MicrosoftNetCoreAppRuntimePackRidDir)', 'native'))</MicrosoftNetCoreAppRuntimePackNativeDir>
+  </PropertyGroup>
```
Comment (ViktorHofer) [inline src/mono/wasm/wasm.proj:16]: Looks great, thanks. This is the only code path that I would like to see changed before we merge the PR. I might be able to contribute to the PR and fix this.

Comment (akoeplinger) [reply src/mono/wasm/wasm.proj]: Updated.

File: src/mono/wasm/wasm.proj [hunk 7]
```diff
@@ -35,32 +50,42 @@
...
-    AssemblyFile="$(GenerateWasmBundlesAssemblyPath)"/>
-  <Target Name="BundleWasm" DependsOnTargets="CheckEnv;LoadTimeZones">
+    AssemblyFile="$(WasmBuildTasksAssemblyPath)"/>
+  <Target Name="BundleTimeZones" DependsOnTargets="CompileTimeZones">
     <PropertyGroup>
       <TimeZonesDataPath>$(NativeBinDir)dotnet.timezones.blat</TimeZonesDataPath>
     </PropertyGroup>
     <GenerateWasmBundle
-      InputDirectory="$(ArtifactsObjDir)wasm/data/output"
+      InputDirectory="$(ArtifactsObjDir)wasm/timezones"
```
Comment (ViktorHofer) [inline src/mono/wasm/wasm.proj:79]: sad panda when you use forward slashes in msbuild 😭

Comment (akoeplinger) [reply src/mono/wasm/wasm.proj:79]: it's the best slash 😸

**Review by ViktorHofer — APPROVED** [review]
